flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/CodeGenUtils.scala

```diff
@@ -195,6 +195,38 @@ object CodeGenUtils {
     case _ => boxedTypeTermForType(t)
   }
 
+  /**
+   * Execute primitive unboxing.
+   */
+  def unbox(term: String, ty: LogicalType): String = ty.getTypeRoot match {
```

Comment:
nit: why in Scala? We could have moved it to CastUtils where also `CastRuleUtils.ternaryOperator` is located?